ios/RCTWebRTC/WebRTCModule+DailyDevicesManager.m

```diff
@@ -179,31 +179,26 @@ - (BOOL)isBuiltInMic:(NSString*)portType {
     // Ducking other apps' audio implicitly enables allowing mixing audio with
     // other apps, which allows this app to stay alive in the backgrounnd during
     // a call (assuming it has the voip background mode set).
-    AVAudioSessionCategoryOptions categoryOptions = (AVAudioSessionCategoryOptionMixWithOthers);
+    // After iOS 16, we must also always keep the bluetooth option here, otherwise
+    // we are not able to see the bluetooth devices on the list
+    AVAudioSessionCategoryOptions categoryOptions = (AVAudioSessionCategoryOptionAllowBluetooth |
+                                                     AVAudioSessionCategoryOptionMixWithOthers);
     NSString *mode = AVAudioSessionModeVoiceChat;
-    
+
     // Earpiece: is default route for a call.
     // Speaker: the speaker is the default output audio for like music, video, ring tone.
     // Bluetooth: whenever a bluetooth device connected, the bluetooth device will become the default audio route.
     // Headphones: whenever any headphones plugged in, it becomes the default audio route even there is also bluetooth device.
     //  And it overwrites the handset(iPhone) option, which means you cannot change to the earpiece, bluetooth.
-    if([WIRED_OR_EARPIECE_DEVICE_ID isEqualToString: deviceId]){
```

Comment:
I'm a bit uneasy about this change because the code was the way it was on iOS < 16 for a reason.

If I'm remembering correctly, we didn't want to provide `AVAudioSessionCategoryOptionAllowBluetooth` in the case that `WIRED_OR_EARPIECE_DEVICE_ID` was selected since it would then cause the phone to prefer using the bluetooth device instead of the earpiece that the user requested.

Reply:
Hi Paul, yeah, I totally see what you mean, I was sharing this same feeling before fixing It.

But if we read the **_AVAudioSessionCategoryOptionAllowBluetooth_** docs, you will see this:
```
@var AVAudioSessionCategoryOptionAllowBluetooth
        Allows an application to change the default behavior of some audio session categories with
        regard to whether Bluetooth Hands-Free Profile (HFP) devices are available for routing. The
        exact behavior depends on the category.

        AVAudioSessionCategoryPlayAndRecord:
            AllowBluetooth defaults to false, but can be set to true, allowing a paired bluetooth
            HFP device to appear as an available route for input, while playing through the
            category-appropriate output.
```

So, we actually need this defined in order for Bluetooth to appear in the list of available audio routes. In theory, based on the docs, we should need this even before iOS 16, but before that, It was working anyway. 

Without this, on Android 16, we don't even receive the notification from iOS when a new bluetooth device is plugged.

And [here](
https://github.com/daily-co/react-native-webrtc/blob/ENG-5575-fix-bluetooth-issue/ios/RCTWebRTC/WebRTCModule%2BDailyDevicesManager.m#L205-L213) is where we are making sure, that despite the fact that bluetooth is available, we are still able to choose the earpiece.

Reply:
I have tested on iOS `15.7.1` and on iOS `16.0.1`, now It is working on both. And looks like the right way to do It, based on the docs.

Reply:
So what happens in the case that I described above? If you do have a bluetooth device connected but explicitly opt to use the earpiece?

Reply:
Aha! @filipi87 pointed out that we weren't relying on omitting bluetooth from the categoryOptions anyway thanks to [this other bit of code](https://github.com/daily-co/react-native-webrtc/blob/ENG-5575-fix-bluetooth-issue/ios/RCTWebRTC/WebRTCModule%2BDailyDevicesManager.m#L205-L213). So we could've (should've) done the refactor here at any point before iOS 16.

Reply:
We choose our preferred input, and that is why It keep working. =)
https://github.com/daily-co/react-native-webrtc/blob/ENG-5575-fix-bluetooth-issue/ios/RCTWebRTC/WebRTCModule%2BDailyDevicesManager.m#L205-L213